products/microos/main.pm

```diff
@@ -68,12 +68,21 @@ sub load_feature_tests {
         loadtest 'console/kubeadm';
     }
     elsif (check_var 'SYSTEM_ROLE', 'container-host') {
-        my $args = OpenQA::Test::RunArgs->new();
-        my $runtime = get_required_var('CONTAINER_RUNTIME');
-        $args->{runtime} = $runtime;
+        my $args_podman = OpenQA::Test::RunArgs->new();
+        my $args_docker = OpenQA::Test::RunArgs->new();
+        $args_podman->{runtime} = 'podman';
+        $args_docker->{runtime} = 'docker';
         loadtest 'microos/toolbox';
+
+        # Podman tests
         load_container_engine_test('podman');
-        loadtest('containers/image', run_args => $args, name => "image_$runtime");
+        loadtest('containers/rootless_podman');
+        loadtest('containers/image', run_args => $args_podman, name => 'image_podman');
+
+        # Docker tests
+        loadtest('microos/install_docker');
```

Comment:
Please get rid of the test module and simply modify `containers::common::install_docker_when_needed`.

Reply:
Since it's a transactional-server specific with a reboot, I would like to leave it as an individual module which serves as a test itself, to see if it's possible to install and fail otherwise.

Reply:
We are already doing transaction server specific actions in the btrfs test module. There is simple task: Install docker. Now there will be two places installing docker and noone will know which of those is used where. Also when your module or any module after your module fails the SUT will restore snapshot where the docker is not yet available. This is already getting messy and too complicated. Please do not create duplicates.